File: go/convert_binary_number_in_linked_list_to_integer.go
package leetcode

type ListNode struct {
	Val  int
	Next *ListNode
}

func getDecimalValue(head *ListNode) int {
	res, curr := 0, head
	for curr != nil {
		res <<= 1
		res |= curr.Val
		curr = curr.Next
	}
	return res
}

File: python/convert-binary-number-in-linked-list-to-integer.py
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next
        
class Solution:
    def getDecimalValue(self, head: ListNode) -> int:
        res, curr = 0, head
        while curr:
            res <<= 1
            res |= curr.val
            curr = curr.next
        return res